File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const brackets = {
        '{':'}',
        '(':')',
        '[':']'
    }
    var stack = [];
    for (let bracket of s.split('')) {
        if (bracket in brackets) {
            stack.push(bracket);
        } else {
            if (brackets[stack.pop()] != bracket) {
                return false;
            }
        }
    }
    if (stack.length > 0) return false;
    return true;
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        # if open add to stack
        # if closing pop from stack and compare
            # if not equal return False
        stack = collections.deque()
        brackets = {')': '(',
                    '}': '{',
                    ']': '['
                   }
        
        for bracket in s:
            if bracket in brackets:
                if stack and stack[-1] == brackets[bracket]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(bracket)
                
        return True if not stack else False
            
        
        